### ISSUE TITLE
BigQuery export results in temp files

### ISSUE THREAD

**Issue by Spikey123** [GoogleCloudPlatform/python-docs-samples#1010]
Following along the Python [BigQuery export docs](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/bigquery/api/load_data_from_csv.py) I was able to automate a process to pull data out of BigQuery and into Google Cloud Storage with the specified poll_job() function.

About 2-3 days ago I started seeing an error come out 

`File "export/export.py", line 265, in main
    raise RuntimeError(msg)
RuntimeError: {u'reason': u'notFound', u'message': u'Not found: Files /bigstore/path//20170628/file1--000000000000.json'}`

That line 265 in the code is right after the call
`success, msg = poll_job(bigquery, job, poll_interval=poll_interval)` from the example

When I check the Google Cloud Storage for the times it worked (I have a bunch of exports in a row and sometimes it works for a few of them) it looks like this: 

![image](https://user-images.githubusercontent.com/24301179/27736993-7e9d6fc8-5d74-11e7-95e0-de8ab56b08b2.png)

I've never seen those temp files there before.

I didn't change anything on my end for a few months. Did anything in the library change that could affect this?

Thank you!

**Comment by tswast** [comment]
This seems likely to be a BigQuery backend issue. I'm checking to see if this is expected behavior.

**Comment by tswast** [comment]
Yes, this is a backend bug. A potential work around is to make sure the job has both WRITE and MODIFY access to the bucket so that BigQuery can clean up temporary files.

**Comment by Spikey123** [comment]
Thank you, @tswast! I'll give it a shot.

When do you think this could potentially get released? Is it more weeks/months/ever? :)

**Comment by nahnhnahk** [comment]
In addition to that, can you try replacing the double slash in the output path with single slash:
So instead of "gs://path//20170628/file1--*.json", use "gs://path/20170628/file1--*.json" instead.

I believe this would help make the job succeed. We have a fix for this as @tswast mentioned and it's being rolled out, but it normally takes about 3 weeks for the rollout to complete.

**Comment by Spikey123** [comment]
Thanks @nahnhnahk that's a good catch. I actually saw it myself as I was pasting the snippet but thought to myself "that can't be an issue given the un-predictable nature of the fails and that it's been working without a hitch for months without changes." That said, I'll go ahead and update that too.

I'll go ahead any attempt to update the service account permissions for WRITE and MODIFY and will wait for the fix to roll out.

Thank you both for the help!
